src/test/java/org/mqttbee/example/Mqtt3SmokeTest.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2018 The MQTT Bee project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.mqttbee.example;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mqttbee.api.mqtt.datatypes.MqttQoS;
+
+import java.io.IOException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class Mqtt3SmokeTest {
+
+    private static final String KEYSTORE_PATH = "src/test/resources/testkeys/mosquitto/mosquitto.org.client.jks";
+    private static final String KEYSTORE_PASS = "testkeystore";
+    private static final String TRUSTSTORE_PATH = "src/test/resources/testkeys/mosquitto/cacerts.jks";
+    private static final String TRUSTSTORE_PASS = "testcas";
+    private final String server = "test.mosquitto.org";
+    private final MqttQoS qos = MqttQoS.AT_LEAST_ONCE;
+    private String topic;
+    private Mqtt3ClientExample subscribeInstance;
+    private Mqtt3ClientExample publishInstance;
+    private final int count = 100;
+
+    @BeforeEach
+    void subscribe() {
+        topic = UUID.randomUUID().toString();
+        subscribeInstance = new Mqtt3ClientExample(server, 1883, false, null, null, null, null);
+        subscribeInstance.subscribeTo(topic, qos).subscribe();
+    }
+
+    @AfterEach
+    void check() {
+        assertEquals(subscribeInstance.getReceivedCount(), publishInstance.getPublishedCount());
```

Comment:
No synchronization with the subscribeInstance
messages are received asynchronous and not waited for here

Reply:
DONE